functions.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * Opens a connection to the collection database
+ * @return object
+ */
+function getDB(): object {
+    return new PDO('mysql:host=db; dbname=jordan-collection', 'root', 'password');
+}
+
+/**
+ * Queries the database and returns selected data as an array
+ * @param object $db - The database that you wish to query
+ * @return array - All matching data will be placed in an array
+ */
+function getData(object $db): array {
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    $query = $db->prepare('SELECT `albumName`, `artistName`, `genre`, `purchaseDate`, `albumImage` FROM `records`;');
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**
+ * Takes an array of collection data and outputs each item to its own <div> container, with formatting
+ * @param array $data - Collection of data fetched from database
+ * @return string
+ */
+function outputRecords(array $data): string {
+    $result = '';
+    // Check that array has required fields
+    if (!count($data)) {
+        return 'You haven\'t added to the collection yet';
+    }
+    foreach ($data as $record) {
+        // If album cover is missing, insert placeholder image
+        $record['albumImage'] = $record['albumImage'] ?? './images/missingCover.png';
```

Comment:
would this work inline when outputting the data? if so it would prevent mutating the data